docs/markdown/snippets/gtkdoc_build_during_install.md

```diff
@@ -0,0 +1,31 @@
+## gnome.gtkdoc new option `build_during_build` can build during build
+
+`gnome.gtkdoc` has had a long outstanding bug where gtk documentation was
+not build during `meson compile` but during `meson install`, leading to a
+wide set of errors. However, simply forcing the gtk documentation to be
+built during `meson compile` is likely going to disrupt many meson users, due
+to the long times required to build gtk documentation. This is unfortunate,
+as users should not rely on meson bugs to simplify their work (in this case,
+reduce build time by skipping documentation builds). The right way for users
+to do this is to create a `gtk_doc` boolean meson option which conditionally
+enables building the documentation:
+
+```meson
+gnome = import('gnome')
+
+if get_option('gtk_doc')
+  gnome.gtkdoc(...)
+endif
```

Comment:
If you wanna get really fancy, you could actually use a `feature` type option and do this (I think):
```meson
gnome = import('gnome', required : get_option('gtk_doc'), disabler : true)
gnome.gtkdoc(...)
```

Reply:
Thanks for the suggestion! I included it, but kept the previous example for people that might do more things than just running `gnome.gtkdoc`.